File: day1.py
with open('./input1.txt') as f:
    food_list = f.read().split('\n')
    food_cal = 0
    cals_list = []
    max_cals_sum = 0

    for food in food_list:
        try:
            food_cal += int(food)
        except:
            cals_list.append(food_cal)
            food_cal = 0

    for i in range(3):
        max_cals_sum += max(cals_list)
        cals_list.remove(max(cals_list))


    print(max_cals_sum)

File: day1.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"sort"
	"strconv"
)

func main() {
    file, err := os.Open("./input1.txt")
    if err != nil {
        panic(err)
    }
    defer file.Close()

    caloriesOfElves := []int{0}

    scanner := bufio.NewScanner(file)
    for scanner.Scan() {  // advances to next token; default: ReadLines
        calorieString := scanner.Text()

        calorieNum, err := strconv.Atoi(calorieString)
        if err != nil {
            caloriesOfElves = append(caloriesOfElves, 0)
            continue
        }
        caloriesOfElves[len(caloriesOfElves) - 1] = caloriesOfElves[len(caloriesOfElves) - 1] + calorieNum
    }

    // Each elem in caloriesOfElves is the total calories of a particular elf.
    sort.Ints(caloriesOfElves)

    // Part 1
    fmt.Println("Max calories among elves:", caloriesOfElves[len(caloriesOfElves)-1])
    // Part 2
    fmt.Println("Max calories among top-three elves:", caloriesOfElves[len(caloriesOfElves)-1] + caloriesOfElves[len(caloriesOfElves)-2] + caloriesOfElves[len(caloriesOfElves)-3])
    // Part 2
}
